### DISCUSSION TITLE
An error occurred with your deployment

### DISCUSSION THREAD

**Discussion by Abhikumar98** [vercel/next.js#31966] [Help]
I'm facing this error suddenly for almost all of my API calls.
Any idea why this might be happening?   
It goes away after redeploying for some but then appears again after some time.

```
code: "504"
message: "An error occurred with your deployment"
```

**Comment by jackHedaya** [comment]
Hey! I think more information is going to be needed for someone to help solve your problem.

Where are you deploying your Next app? What do the runtime and deployment logs say? Hoping are you calling the API and how are are the working routes receiving the request vs the broken routes?

**Reply by Abhikumar98** [reply]
Hey thanks for answering, I went ahead looking for more info in logs and found out that the APIs are being timed out.
Not sure why though cause they were working fine until a couple of days ago and then randomly started failing.

I've checked in multiple routes and found out that the async calls inside the routes finish up within 5s max but don't send a response back in time.

Wondering if it's due to the recent issue. https://www.vercel-status.com/

**Reply by handy-castro** [reply]
@Abhikumar98 I am also having the same issue. Have you been able to get around it?

**Reply by Abhikumar98** [reply]
Redeploying solved this for me. Also, there was a timeout error, even though the logic inside the function was taking only 2 - 3 secs max. the response was returned after 10 secs which was causing the timeout error.
I think it was momentary as it's not happening right now.
They've mentioned an issue with serverless functions in https://www.vercel-status.com and told that redeploying in another region would solve it ( I don't think we have the option to that though )

**Comment by Hamza0302** [comment]
This issue will be also created when you are api dint have res.send method . I was testing and forget to send the res back to client.

**Reply by RomneyDa** [reply]
this solved it for me! To be specific I was using:
```typescript
return new Response("Unauthorized", { status: 401 })
```

and changing to this (with NO return) fixed it:
```typescript
res.status(401).send("Unauthorized")
```

[docs](https://nextjs.org/docs/pages/building-your-application/routing/api-routes)

**Comment by lucasuema** [comment]
> I'm facing this error suddenly for almost all of my API calls. Any idea why this might be happening? It goes away after redeploying for some but then appears again after some time.
> 
> ```
> code: "504"
> message: "An error occurred with your deployment"
> ```

I'm facing the same issue. Any way to fix it? Re-deploying fixes the issue but after a few hours or a day the issue starts happening again! 🤕

**Reply by meel-hd** [reply]
[Vercel docs](https://vercel.com/docs/concepts/functions/serverless-functions#execution-timeout) explains this error.

**Reply by AbhayaShankar** [reply]
I was getting the same error as everybody else

I went through the vercel docs of [serverless functions execution timeout](https://vercel.com/docs/concepts/functions/serverless-functions#execution-timeout) and I wanted to ask or rather clarify how to use vercel.json file as described in the docs

![image](https://github.com/vercel/next.js/assets/113185442/1d5f85ae-c9e0-42bb-9aed-247ef7ee9f2d)

And if I don't have teams account rather using a personal account then how can I rectify this?

**Comment by MattinaYang** [comment]
The 504 error suddenly occurs while the api worked with no problem before, and redeploying is not working for me :(
Waiting for a method to solve this problem.
![image](https://user-images.githubusercontent.com/78125933/223690470-a86d46b2-b71d-437c-af41-0b1510370f4b.png)

**Reply by chezhe** [reply]
I got this error when I request the API with custom headers.

**Reply by KhoeLe** [reply]
I had a problem and my problem was solved when I retry calling my API. Let me explain in images blow the status pre = "building" and I retry call API and then status update = "completed"
![image](https://user-images.githubusercontent.com/63848705/230135424-c992c09a-782b-430f-a0fe-8c57d510535c.png)

**Reply by qizeng19** [reply]
how to  solved it ?

**Reply by poodaydreamer** [reply]
I am having same issue.

It was working a couple of days ago but now, I suddenly started getting 504 issue.
<img width="1117" alt="image" src="https://github.com/vercel/next.js/assets/49583931/4a273798-ea2e-4233-8850-70ba06ad96f8">


Is it related to Vercel system status?

**Reply by real-jiakai** [reply]
## Updates

I found the reason: https://stackoverflow.com/a/71485015/16720823

> When using Vercel with a Hobby plan, your serverless API routes can only be processed for ~~5 seconds~~（now limit for hobby plan is 10s). This means that after ~~5 seconds~~(10s), the route responds with a 504 GATEWAY TIMEOUT error.

Maybe vercel serveless api routes are slow today and  the time of interacting with my chatgpt api is over 10 s.

## My issue

I also encountered this trouble in vercel. A few days ago, my application was running normally.

These days, I have not developed, but when I tested my application today, I found that the API routing error.

![image](https://github.com/vercel/next.js/assets/82650452/42e9bdc8-d30a-4f2d-8eb7-9e166b2fbb4e)

![image](https://github.com/vercel/next.js/assets/82650452/d21575f7-6e69-43f8-878b-8781e021817a)

504 error.

I use api route in localhost and it works. But when I deploy my app to vercel, it shows 504 error.

Now I deployed my app to Netlify and still got 5xx error.

The netlify free limit time is 10s. Maybe you can try. In my case, I must self hosted my nextjs app.

**Reply by Yasssh-bombale** [reply]
how you selfhost your nextjs app?? plz reply

**Reply by real-jiakai** [reply]
Yep, I use docker to deploy my Next.js app. Nowadays Vercel function limit time increases to 60s, it should be enough for personal use.

**Comment by MarvingMoreton** [comment]
Yup, had the same issue with GPT API, implementing a vercel.json file at the root level definitely helped.

https://vercel.com/docs/concepts/functions/serverless-functions#execution-timeout

**Comment by cngvc** [comment]
In my case: (nodejs, typescript)
1. `vercel pull`
2. create a `vercel.json` at root project
```
{
  "version": 2,
  "name": "project-id-on-vercel-dashboard",
  "public": true,
  "builds": [{ "src": "src/index.ts", "use": "@vercel/node" }],
  "routes": [{ "src": "/(.*)", "dest": "/src/index.ts" }]
}

```
3. `vercel --prod`

**Comment by xArrietAx** [comment]
Yesterday I had this same problem, and the only thing I did was to continue using Render, as that's where I did my first deployment (I'm a beginner). I followed these steps: https://render.com/docs/deploy-nextjs-app. For anyone doing the same, read about the version of Node.js that Render uses, and change the Node.js version if your application requires it: https://render.com/docs/node-version, as I got a mongoose error yesterday because I hadn't read it.

**ESPAÑOL**

Ayer tenía este mismo problema y lo unico que hice fue seguir utilizando render, ya que ahí hice mi primer deploy (Soy un principiante). Seguí los siguiente pasos: https://render.com/docs/deploy-nextjs-app .  Para el que vaya a hacer lo mismo, lean sobre la version de node js que usa render y cambien la version de node js si su aplicación lo requiere: https://render.com/docs/node-version ya que ayer me dio error de mongoose por no haber leido.

**Comment by LeoUpperThrower4** [comment]
I had the same issue. 
I could fix it by **stop fetching a Nextjs API Route inside another Nextjs API Route**. I believe that happens because of [cold starts](https://vercel.com/docs/functions/serverless-functions#improving-serverless-function-performance) from serverless functions.

I changed from:
```js
export default async function handler(req: NextApiRequest, res: NextApiResponse) {
  if (req.method == 'POST') {
    axios.get(`${process.env.NEXT_PUBLIC_BASE_URL}/api/ANOTHER-ROUTE`)
  }
}
```
To:
```js
export default async function handler(req: NextApiRequest, res: NextApiResponse) {
  if (req.method == 'POST') {
    // db.query [...]
    // Do database operations instead of fetching another local route
  }
}
```


The first API call triggers the first serverless function, which then triggeres another serveless funcion (since we are calling another route). That means we are using Cold Start Time + Execution Time of our second function in the short limit ([15s in Vercel's hobby plan](https://vercel.com/docs/functions/serverless-functions/runtimes#maxduration)) of our first function.

**Comment by surajbahadur591** [comment]
I am using openai free trial and for me the error was mainly because the reponse time was more than 10s.
![image](https://github.com/vercel/next.js/assets/36602276/1e25f468-0ed0-4279-b254-a6350551c3ba)

**Reply by AbhayaShankar** [reply]
Yes vercel free tier doesn't allow long requests. For that you would have to buy its pro models.
You can read about that [here](https://vercel.com/docs/accounts/plans/hobby)

![image](https://github.com/vercel/next.js/assets/113185442/58e255cb-d0e2-4a40-bb96-a8cbcb0270b9)

**Comment by Gijela** [comment]
so? what's the answer of the question?

**Reply by AbhayaShankar** [reply]
Either switch to a paid premium plan. Or else keep trying to de-bug the error IG 🙃🥲

**Comment by Yasssh-bombale** [comment]
but in my case it still throwing error how can i update this function limit
to 60s

**Comment by real-jiakai** [comment]
> but in my case it still throwing error how can i update this function limit to 60s
> […](#)
> On Sun, May 19, 2024 at 9:07 PM Jaya ***@***.***> wrote: Yep, I use docker to deploy my Next.js app. Nowadays Vercel function limit time increases to 60s, it should be enough for personal use. — Reply to this email directly, view it on GitHub <[#31966 (reply in thread)](https://github.com/vercel/next.js/discussions/31966#discussioncomment-9488648)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/A5RB5TQ3PSBDHU3O7NSDUKLZDDBJZAVCNFSM5JCKBYFKU5DIOJSWCZC7NNSXTOKENFZWG5LTONUW63SDN5WW2ZLOOQ5TSNBYHA3DIOA> . You are receiving this because you commented.Message ID: <vercel/next. ***@***.***>

you can find more information in this page: [What can I do about Vercel Functions timing out?](https://vercel.com/guides/what-can-i-do-about-vercel-serverless-functions-timing-out)

**Comment by Yasssh-bombale** [comment]
It works! Thanks a lot 😅

**Comment by QastanFIT** [comment]
Still this error. I am not using ChatGPT. I am fetching data from another API (http). It does work in localhost, but not in production. The results I am getting from my API are in like 200-500ms or so... So no where near the 5-10s limit
